modules/lib/shell.nix

```diff
@@ -41,6 +41,83 @@ rec {
   # statement for each set entry.
   exportAll = vars: lib.concatStringsSep "\n" (lib.mapAttrsToList export vars);
 
+  # Wrap a list of strings to a given line width.
+  # Packs as many items as possible per line without exceeding maxWidth.
+  # Returns a list of strings, each representing a line.
+  #
+  # Example: wrapLines ["item1" "item2" "very-long-item" "item3"] 20
+  #   => ["item1 item2" "very-long-item" "item3"]
+  wrapLines =
+    items: maxWidth:
+    let
+      step =
+        acc: item:
+        let
+          potentialLine = if acc.currentLine == "" then item else "${acc.currentLine} ${item}";
+        in
+        if lib.stringLength potentialLine <= maxWidth then
+          acc // { currentLine = potentialLine; }
+        else
+          acc
+          // {
+            finishedLines = acc.finishedLines ++ [ acc.currentLine ];
+            currentLine = item;
+          };
+      foldResult = lib.foldl' step {
+        finishedLines = [ ];
+        currentLine = "";
+      } items;
+    in
+    foldResult.finishedLines ++ lib.optional (foldResult.currentLine != "") foldResult.currentLine;
+
+  # Format a list of lines into an indented, multi-line block.
+  # Adds 2-space indentation and wraps with newlines for shell array formatting.
+  #
+  # Example: formatMultiLine ["line1" "line2 line3"]
+  #   => "\n  line1\n  line2 line3\n"
+  #
+  # Designed for shell arrays but reusable for any indented multi-line output.
+  formatMultiLine = lines: "\n  ${lib.concatStringsSep "\n  " lines}\n";
```

Comment:
This feels like an implementation detail rather than an actual library function IMO.

Reply:
Yeah.. just wanted to try to break it up to be composable but probably not helpful.